web/WEB-INF/web.xml

```diff
@@ -14,9 +14,41 @@
     <description>An address where OpenGrok can receive new configuration</description>
   </context-param>
 
+  <!--
+  <context-param>
+      <param-name>authorizationPluginDirectory</param-name>
+      <param-value>/var/opengrok/plugins</param-value>
+      <description>
+          Directory with authorization plugins.
+          Default value is dataRoot/../plugins 
+            (can be /var/opengrok/plugins if
+             dataRoot is /var/opengrok/data)
+      </description>
+  </context-param>
+  
+  <context-param>
+      <param-name>enableAuthorizationWatchDog</param-name>
+      <param-value>true</param-value>
+      <description>
+          Enable watching the plugin directory for changes in real time.
+          Suitable for development.
+      </description>
+  </context-param>
+  -->
+  
   <listener>
    <listener-class>org.opensolaris.opengrok.web.WebappListener</listener-class>
   </listener>
+
```

Comment:
This needs to be as static as possible. Avoid new object creation, cache everything that can be cached.
Get internally how many requests we get per minute on grok.cz @vladak has the logs. This should be the lowest target for scaling, nothing less.